armi/reactor/converters/uniformMesh.py

```diff
@@ -85,6 +86,321 @@ def converterFactory(globalFluxOptions):
+
+        Returns
+        -------
+        filteredBottomCtrl : the bottom of control assemblies, filtered to a minimum separation of
+            ``minimumMeshSize`` with preference for the lowest bounds
+        filteredTopCtrl : the top of control assemblies, filtered to a minimum separation of
+            ``minimumMeshSize`` with preference for the top bounds
+        """
+        bottomMatBoundaries = set(fuelBottoms)
+        topMatBoundaries = set(fuelTops)
+
+        # find all control assembly boundaries
+        for a in self._sourceReactor.core.getAssemblies(Flags.CONTROL):
+            firstBlock = a.getFirstBlock(Flags.CONTROL)
+            lastBlock = a.getBlocks(Flags.CONTROL)[-1]
+            bottomMatBoundaries.add(firstBlock.p.zbottom)
+            topMatBoundaries.add(lastBlock.p.ztop)
+
+        bottomBoundList = sorted(list(bottomMatBoundaries))
+        topBoundList = sorted(list(topMatBoundaries))
+        # filter control boundaries to minimum mesh size
+        filteredBottomCtrl = self._filterMesh(
+            bottomBoundList,
+            self.minimumMeshSize,
+            fuelBottoms,
+            preference="bottom",
+            warn=True,
+        )
+        filteredTopCtrl = self._filterMesh(
+            topBoundList, self.minimumMeshSize, fuelTops, preference="top", warn=True
+        )
+        return filteredBottomCtrl, filteredTopCtrl
+
+    @staticmethod
+    def _determineIndexToRemove(ileft, iright, meshPoints, anchorPoints):
+        """
+        Determine which mesh to remove when two points are too close together
+
+        Rules:
+            1. Always keep anchorPoints
+            2. If neither mesh point is in anchorPointws, keep the "left" one
+
+        Notes
+        -----
+        ``UniformMeshGenerator._filterMesh`` sorts the mesh based on the option
+        passed for ``preference`` (``bottom`` or ``top``) so in this method,
+        ``ileft`` always has preference over ``iright``.
+
+        Parameters
+        ----------
+        ileft : int, required
+            "left" index (bottom if preference is "bottom", top if preference is "top"
+        iright : int, required
+            "right" index (opposite of the "left" index)
+        meshPoints : [float, float], required
+            The two mesh points within minimumMeshSize of each other
+        anchorPoints : list[float], required
+            The anchor points that will not be removed from the mesh
+        """
+        meshLeft, meshRight = tuple(meshPoints)
+        if meshRight in anchorPoints:
+            if meshLeft in anchorPoints:
+                runLog.error(
+                    "Attempting to remove two anchor points!\n"
+                    "The uniform mesh minimum size for decusping may be smaller than the "
+                    "gap between anchor points:\n"
+                    f"{meshLeft}, {meshRight}, gap = {abs(meshLeft-meshRight)}"
+                )
+                raise ValueError
```

Comment:
As is common around these parts, you added this exception, but it is not covered by unit tests.